tinygrad/codegen/linearizer.py

```diff
@@ -133,10 +133,16 @@ def __init__(self, ast:LazyOp, output_buffer:LazyBuffer):
     # get the output buffers
     self.bufs = [output_buffer] + dedup(ast.buffers)
 
+    # dedup by raw buffer
+    self.dedup_bufs = dedup([x.realized for x in self.bufs], self.bufs)
```

Comment:
What is x.realized is None?

Reply:
When can this happen? I was under the impression that inputs to Linearizer must be realized. What does a realized buffer with .realized == None mean?